userspace/libscap/engine/gvisor/scap_gvisor.cpp

```diff
@@ -86,33 +93,53 @@ engine::~engine()
 
 }
 
-int32_t engine::init(std::string socket_path)
+int32_t engine::init(std::string socket_path, std::string runsc_root_path, std::string trace_session_config_path)
 {
-	// Initialize the listen fd
-	int sock, ret;
-	struct sockaddr_un address;
-	unsigned long old_umask;
 	m_socket_path = socket_path;
-	if (m_socket_path.empty())
+	if(m_socket_path.empty())
 	{
 		strlcpy(m_lasterr, "Empty gVisor socket path", SCAP_LASTERR_SIZE);
 		return SCAP_FAILURE;
 	}
-
 	unlink(m_socket_path.c_str());
```

Comment:
Just to remember it:`unlink` syscall doesn't exist in ARM64, maybe we can add a comment here to remember it

Reply:
Mmmh this is passing build-arm64 ci though.

Btw we **cannot** break arm64 build (that's why we got the CI :D )

Reply:
The `libc` wrapper is there, the problem is that under the hood the `SYS_unlink` syscall doesn't exist in ARM64, so I would avoid calling it, instead, I will call directly  `syscall(SYS_unlinkat, ...); `. This is just my opinion, and must not be done in this PR, but I wrote it here just to remember it. Same for all other syscalls that I have highlighted in this review